plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -688,20 +688,18 @@ public void finishInsertTable(ConnectorSession session, JdbcOutputTableHandle ha
                 .map(this::quoted)
                 .collect(joining(", "));
         String insertSql = format("INSERT INTO %s (%s) SELECT %s FROM %s", targetTable, columnNames, columnNames, temporaryTable);
-        String cleanupSql = "DROP TABLE " + temporaryTable;
 
         try (Connection connection = getConnection(session, handle)) {
             execute(connection, insertSql);
         }
         catch (SQLException e) {
             throw new TrinoException(JDBC_ERROR, e);
         }
-
-        try (Connection connection = getConnection(session, handle)) {
-            execute(connection, cleanupSql);
-        }
-        catch (SQLException e) {
-            log.warn(e, "Failed to cleanup temporary table: %s", temporaryTable);
+        finally {
+            JdbcTableHandle temporaryTableHandle = new JdbcTableHandle(
+                    new SchemaTableName(handle.getSchemaName(), handle.getTemporaryTableName()),
+                    new RemoteTableName(Optional.ofNullable(handle.getCatalogName()), Optional.ofNullable(handle.getSchemaName()), handle.getTemporaryTableName()));
+            dropTable(session, temporaryTableHandle);
```

Comment:
I think you need another method that accepts `RemoteTableName` and then it would be nice to use `RemoteTableName` in `JdbcOutputTableHandle`, so you could simply `dropTable(handle.getTemporaryTableName()`